argo-cd-apps/base/host/smee/kustomization.yaml

```diff
@@ -0,0 +1,6 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```

Comment:
While testing this change in my dev environment for the namespace question, I realized that you are missing to include this folder/kustomization.yaml in `argo-cd-apps/base/host/kustomization.yaml`, without this, the application is not created for the host

Reply:
I only want this in stage for now, I'll more it to base once I test it for a while.

I think using delete-application is overkill at this early phase.

Reply:
ok, but if you do not add it in `argo-cd-apps/base/host/kustomization.yaml`, the application will not even be in staging

Reply:
If I build the staging overlay, the smee application is not there:
```
kustomize build argo-cd-apps/overlays/staging/ | grep smee
```

Reply:
Just a note, without the delete application patch it will get deployed to all environments, not just staging.

Reply:
I just tested and it is deployed to dev from staging component folder, it will also be deployed to prod from staging folder. I unresolved the thread so you see it. If you want to proceed as is, let me know and I will resolve the thread and put lgtm label.

Reply:
@hugares Are you seeing the resources get deployed or just empty applications? I don't want the resources to be deployed in production, and I don't particularly care about dev ATM. 

@gbenhaim I thought only having stuff in `staging` directory under the component would be sufficient, otherwise what is the point of having those directories and all the complex Application resource that lead to them?

Reply:
Ok, I think I've found what you meant - pade sure to apply the overlay patch in production and the deletion patch in dev.